tests/pipeline/test_pipeline_state.py

```diff
@@ -503,3 +513,15 @@ def test_migrate_state(test_storage: FileStorage) -> None:
     p = dlt.attach(pipeline_name="debug_pipeline", pipelines_dir=test_storage.storage_path)
     assert p.dataset_name == "debug_pipeline_data"
     assert p.default_schema_name == "example_source"
+
+    # specifically check destination v2 to v3 migration
+    state_v2 = {"destination": "redshift", "staging": "filesystem", "_state_engine_version": 2}
+    migrate_state(
+        "test_pipeline", state_v2, state_v2["_state_engine_version"], STATE_ENGINE_VERSION  # type: ignore
+    )
+    assert state_v2["destination_name"] == "redshift"
+    assert state_v2["destination_type"] == "redshift"
```

Comment:
this looks wrong. it must always be a full path to a module that can be imported.